pkg/logs/launchers/integration/launcher_test.go

```diff
@@ -137,17 +170,274 @@ func (suite *LauncherTestSuite) TestIntegrationLogFilePath() {
 
 	id = "1234 5678:myIntegration"
 	actualFilePath = suite.s.integrationLogFilePath(id)
-	expectedFilePath = filepath.Join(suite.s.runPath, "1234-5678_myIntegration.log")
+	expectedFilePath = filepath.Join(suite.s.runPath, "1234 5678_myIntegration.log")
 	assert.Equal(suite.T(), expectedFilePath, actualFilePath)
 }
 
+// TestFileNameToID ensures file names are decoded to their proper id
+func (suite *LauncherTestSuite) TestFileNameToID() {
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{"file_name_1234.log", "file_name:1234"},
+		{"example_test_5678abcd.log", "example_test:5678abcd"},
+		{"integration with spaces_5678.log", "integration with spaces:5678"},
+		{"file_with_multiple_underscores_9999.log", "file_with_multiple_underscores:9999"},
+	}
+
+	for _, tt := range tests {
+		suite.T().Run(tt.input, func(_ *testing.T) {
+			result := fileNameToID(tt.input)
+			assert.Equal(suite.T(), tt.expected, result)
+		})
+	}
+}
+
+// TestFileExceedsSingleFileLimit ensures individual files cannot exceed file
+// limit sizes
+func (suite *LauncherTestSuite) TestFileExceedsSingleFileLimit() {
+	oneMB := int64(1 * 1024 * 1024)
+	suite.s.combinedUsageMax = 2 * oneMB
+	suite.s.fileSizeMax = oneMB
+
+	filename := "sample_integration_123.log"
+	file, err := os.Create(filepath.Join(suite.s.runPath, filename))
+	assert.Nil(suite.T(), err)
+
+	file.Write(make([]byte, oneMB))
+	file.Close()
+
+	suite.s.Start(nil, nil, nil, nil)
+
+	integrationLog := integrations.IntegrationLog{
+		Log:           "sample log",
+		IntegrationID: "sample_integration:123",
+	}
+
+	suite.s.receiveLogs(integrationLog)
+
+	assert.Equal(suite.T(), int64(len(integrationLog.Log)+1), suite.s.combinedUsageSize)
+	assert.Equal(suite.T(), int64(len(integrationLog.Log)+1), suite.s.integrationToFile["sample_integration:123"].size)
+	assert.Equal(suite.T(), 1, len(suite.s.integrationToFile))
+}
+
+// TestScanInitialFiles ensures files already present in the runPath for the
+// launcher are detected and managed upon launcher start
+func (suite *LauncherTestSuite) TestScanInitialFiles() {
```

Comment:
Should we add a test case that ensure if the initial combined size is exceed that we remove files from `integrationToFile`?

Reply:
Makes sense, I think the `TestInitialLogsExceedTotalUsageMultipleFiles` covers that case 👍